sound/soc/sof/Kconfig

```diff
@@ -179,6 +182,7 @@ config SND_SOC_SOF
 	tristate
 	select SND_SOC_TOPOLOGY
 	select SND_SOC_SOF_NOCODEC if SND_SOC_SOF_NOCODEC_SUPPORT
+	select SND_SOC_SOF_AUDIO
```

Comment:
Ah that's an interesting one.
Do we *always* require a separate platform device, or do we allow in some cases the hierarchy to remain with a single device?
With all the work you've done, I am not sure if we actually require the platform device in all cases.

It's the same e.g. as workqueues for probe, it's required for HDAudio but not for all platforms.